### PR TITLE
Master

### PR DESCRIPTION
**Pull request by grishmashrestha**

### CODE REVIEW

File: app/src/main/java/com/lftechnology/unito/Unito.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.lftechnology.unito;
+
+import android.app.Application;
+import android.content.Context;
+
+import timber.log.Timber;
+
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/Unito.java]: @grishmashrestha Please add class description

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+import com.lftechnology.unito.constant.AppConstant;
+import com.lftechnology.unito.fragment.MainFragment;
+import com.lftechnology.unito.helper.OnStartDragListener;
+import com.lftechnology.unito.helper.SimpleItemTouchHelperCallback;
+import com.lftechnology.unito.utils.GeneralUtils;
+import com.lftechnology.unito.utils.SoftKeyBoard;
+import com.squareup.otto.Subscribe;
+
+import java.util.Arrays;
+
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha same here

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    private DrawerLayout mDrawerLayout;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    private RecyclerView mRecyclerView;
+    private DrawerRecyclerViewAdapter mAdapter;
+    private RecyclerView.LayoutManager mLayoutManager;
+    private String[] mDrawerRecyclerViewDataset;
+    private ItemTouchHelper mItemTouchHelper;
+    private LinearLayout linearLayout;
+    private int DY = 5;
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha why is this variable name capital?

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        super.onPause();
+        EventBus.unregister(this);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mToolbarContainer = (LinearLayout) findViewById(R.id.toolbarContainer);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha why are you initializing views the old fashioned way? I saw butter knife on your dependencies. Use that.

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+    private void setNavigationDrawer() {
+        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
+        mDrawerToggle = setupDrawerToggle();
+        mDrawerLayout.addDrawerListener(mDrawerToggle);
+    }
+
+    private String[] getDrawerRecyclerViewDataset() {
+        String[] drawerRecyclerViewDataset;
+        SharedPreferences sharedPref = getBaseContext().getSharedPreferences("Unito", Context.MODE_PRIVATE);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha move the hard coded strings either to strings.xml. Better create a class and define all the constants there.

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+        setHeaderTextByFragment();
+        setRecyclerView();
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> parent) {
+    }
+
+    public void swapFragments(View view) {
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha please comment on not "easy understood" methods.

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+        ImageView swapButton = (ImageView) findViewById(R.id.swapButton);
+        animateSwapButton(swapButton);
+
+        EventBus.post(new SwapFragment(true));
+    }
+
+    private void animateSwapButton(ImageView swapButton) {
+        Animation an;
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha an is not an appropriate instance name.

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+        if (spinDirection) {
+            an = new RotateAnimation(0.0f, 180.0f, swapButton.getWidth() / 2, swapButton.getHeight() / 2);
+        } else {
+            an = new RotateAnimation(180.0f, 0.0f, swapButton.getWidth() / 2, swapButton.getHeight() / 2);
+        }
+        spinDirection = !spinDirection;
+        an.setDuration(300);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha create a final static variable for 300.

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+    private LinearLayout mToolbarContainer;
+    private String mSelectedConversion;
+    private DrawerLayout mDrawerLayout;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    private RecyclerView mRecyclerView;
+    private DrawerRecyclerViewAdapter mAdapter;
+    private RecyclerView.LayoutManager mLayoutManager;
+    private String[] mDrawerRecyclerViewDataset;
+    private ItemTouchHelper mItemTouchHelper;
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha some variables are mPrefixed and some are camelCased. make it uniform

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+                    drawerRecyclerViewDataset = getResources().getStringArray(R.array.weight_options);
+                    break;
+                default:
+                    drawerRecyclerViewDataset = new String[]{};
+                    break;
+            }
+        } else {
+            Gson gson = new Gson();
+            String[] arrayList = gson.fromJson(preference, String[].class);
+            drawerRecyclerViewDataset = arrayList;
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha drawerRecyclerViewDataset = gson.fromJson(preference, String[].class); i think this does the job too. why extra array initialization.

File: app/src/main/java/com/lftechnology/unito/adapter/DrawerRecyclerViewAdapter.java [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+        Collections.swap(mDataset, fromPosition, toPosition);
+        notifyItemMoved(fromPosition, toPosition);
+        updateUserPreference();
+        return true;
+    }
+
+    private void updateUserPreference() {
+        Gson gson = new Gson();
+        String jsonDataset = gson.toJson(mDataset);
+        SharedPreferences sharedPref = Unito.getContext().getSharedPreferences("Unito", Context.MODE_PRIVATE);
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/adapter/DrawerRecyclerViewAdapter.java]: @grishmashrestha same as above

File: app/src/main/java/com/lftechnology/unito/bus/ConvertedValue.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
+package com.lftechnology.unito.bus;
+
+/**
+ * Created by Grishma Shrestha <grishmashrestha@lftechnology.com> on 3/2/16.
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/bus/ConvertedValue.java]: @grishmashrestha class desc

File: app/src/main/java/com/lftechnology/unito/constant/AppConstant.java [hunk 13]
```diff
@@ -0,0 +1,71 @@
+package com.lftechnology.unito.constant;
+
+/**
+ * Created by Grishma Shrestha <grishmashrestha@lftechnology.com> on 3/3/16.
+ */
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/constant/AppConstant.java]: @grishmashrestha move those hard coded strings here

File: app/src/main/res/layout/fragment_slider_top.xml [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+        <com.lftechnology.unito.utils.CustomEditText
+            android:id="@+id/from_unit"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center"
+            android:background="@android:color/white"
+            android:inputType="numberDecimal"
+            android:maxLength="12"
+            android:minWidth="50dp"
```
Comment (manas-raj-shrestha) [inline app/src/main/res/layout/fragment_slider_top.xml]: @grishmashrestha move all the dimes to dimens.xml

File: app/src/main/java/com/lftechnology/unito/fragment/ScreenSlideTopFragment.java [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+        if (mPosition == 0) {
+            new Handler().postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    EventBus.post(new ConvertedValue(mFromUnit.getText().toString().trim(), mDataset[mPosition], mPosition, mSelectedConversion));
+                }
+            }, 2);
+        }
+
+        mFromUnit.addTextChangedListener(new TextWatcher() {
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/unito/fragment/ScreenSlideTopFragment.java]: @grishmashrestha check out textChangeListener in butterknife
http://jakewharton.github.io/butterknife/javadoc/

File: app/src/main/AndroidManifest.xml [hunk 16]
```diff
@@ -3,21 +3,28 @@
...
+            android:windowSoftInputMode="adjustResize">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <!-- ATTENTION: This was auto-generated to add Google Play services to your project for
+             App Indexing.  See https://g.co/AppIndexing/AndroidStudio for more information. -->
+        <meta-data
```
Comment (shilu-stha) [inline app/src/main/AndroidManifest.xml:24]: @grishmashrestha any purpose to add this? Also fix in dependency.

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 17]
```diff
@@ -0,0 +1,312 @@
...
+import java.util.Arrays;
+
+public class MainActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener, OnStartDragListener, OnKeyEvents {
+    private Toolbar toolbar;
+    private LinearLayout mToolbarContainer;
+    private String mSelectedConversion;
+    private DrawerLayout mDrawerLayout;
+    private ActionBarDrawerToggle mDrawerToggle;
+
+    private RecyclerView mRecyclerView;
```
Comment (shilu-stha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha these fields can be made local. Inspect the code once.

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 18]
```diff
@@ -0,0 +1,312 @@
...
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    private void setSpinner() {
+        Spinner spinner = (Spinner) findViewById(R.id.unito_option_spinner);
```
Comment (shilu-stha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha Why are we using spinners for menu?

File: app/src/main/java/com/lftechnology/unito/activity/MainActivity.java [hunk 19]
```diff
@@ -0,0 +1,312 @@
...
+    @Override
+    public void onStartDrag(RecyclerView.ViewHolder viewHolder) {
+        mItemTouchHelper.startDrag(viewHolder);
+    }
+
+    @Subscribe
+    public void toggleToolBar(ScrollListener scrollListener) {
+        int dy;
+        if (scrollListener.moveUp) {
+            if (GeneralUtils.convertPixelsToDp(mToolbarContainer.getTranslationY(), this) > -56.0) {
```
Comment (shilu-stha) [inline app/src/main/java/com/lftechnology/unito/activity/MainActivity.java]: @grishmashrestha Need to give a proper representation for value "-56.0".

File: app/build.gradle [hunk 20]
```diff
@@ -22,6 +22,13 @@ android {
...
     testCompile 'junit:junit:4.12'
-    compile 'com.android.support:appcompat-v7:23.1.1'
-    compile 'com.android.support:design:23.1.1'
+    compile 'com.android.support:appcompat-v7:23.2.1'
+    compile 'com.android.support:design:23.2.1'
+    compile 'com.jakewharton.timber:timber:4.1.1'
+    compile 'com.jakewharton:butterknife:7.0.1'
+    compile 'com.android.support:support-v4:23.2.1'
+    compile 'com.squareup:otto:1.3.8'
+    compile 'com.google.android.gms:play-services-appindexing:8.1.0'
```
Comment (shilu-stha) [inline app/build.gradle]: @grishmashrestha check this dependency usability.

File: app/src/main/java/com/lftechnology/Dunite/activity/MainActivity.java [hunk 21]
```diff
@@ -0,0 +1,327 @@
...
+        return true;
+    }
+
+    private void setHeaderTextByFragment() {
+        mTv.setText(mSelectedConversion);
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        Spinner spinner = (Spinner) parent;
```
Comment (shilu-stha) [inline app/src/main/java/com/lftechnology/Dunite/activity/MainActivity.java]: @grishmashrestha Why is spinner used for menu?

**Pull request merged**